### PR TITLE
Change Notify icon to NotifyCircle

### PR DESCRIPTION
**Pull request by amelako**
## Purpose

Change the notify icon to  "notify_circle" icon for the yellow variant, and have the element inside in black.
## Approach and changes

_Describe how you solved the problem_

Updated NotificationInline, NotificationToast and ValidationHint components to use the NotifyCircle icon.
Update the warning text color to n80.

## Definition of done

* [ ] Development completed
* [ ] Reviewers assigned
* [ ] Unit and integration tests
* [ ] Meets minimum browser support
* [ ] Meets accessibility requirements

### CODE REVIEW

File: packages/circuit-ui/components/NotificationInline/NotificationInline.tsx [hunk 1]
```diff
@@ -155,16 +155,37 @@ const actionButtonStyles = ({ theme }: StyleProps & ButtonProps) =>
...
+        background: ${theme.colors.black};
+        border-radius: ${theme.borderRadius.circle};
+      }
+
+      svg {
+        position: relative;
+        z-index: 1;
+      }
+    `,
 );
```
Comment (amelako) [inline packages/circuit-ui/components/NotificationInline/NotificationInline.tsx:189]: First of we didn’t have to add a new icon, we already have the NotifyCircle in the icons package.
The first approach we tried to change the background color of the svg and cut out the borders, which didn’t work out. The next idea was to use pseudo elements, but it turns out the svg is not supporting pseudo elements.
The last approach was just creating an IconWrapper as a styled div, and adding pseudo element to just add a black background behind the SVG icon to color just the exclamation mark black.

We would like your thoughts and ideas and @robinmetral. It bring also another topic what in case we have to do some of the changes like this more often.

Comment (robinmetral) [reply packages/circuit-ui/components/NotificationInline/NotificationInline.tsx:189]: I think it makes sense 👍 that's also the implementation I had in mind.

What I'd do is create a `NotifyIcon` component (internal to Circuit) that we can reuse across all components. If we notice that teams also need it we could also export it. WDYT?

Comment (connor-baer) [reply packages/circuit-ui/components/NotificationInline/NotificationInline.tsx:189]: We thought about that too. However, as Amela pointed out, we might face the same challenge with other icons. Whenever an icon has a color that doesn't meet minimum color contrast requirements, we need a way to turn the shape inside black. The approach here is clunky but works for circular and rectangular icons. What about triangular or oddly shaped icons though? 

Ideally, we'd have a way to change the color of a part of an icon separately from its primary color. This would be possible if, e.g. the exclamation mark of the notify icon was its own path that can be targeted with CSS. The downside here is that the exclamation mark couldn't be transparent anymore. 

Maybe I'm overthinking this?

Comment (robinmetral) [reply packages/circuit-ui/components/NotificationInline/NotificationInline.tsx:189]: TBH I think that this should be the concern of a team wanting to use custom colors with icons. Just like when using colors anywhere else, they'd need to check contrast. In most cases icons should be the same as the text color so should be 4.5:1 or higher (although icons can actually go down to 3:1, like large text).

I'm not even sure if technically making the `!` black would pass an audit. Users will probably see it more, but the background is still white, really.

So long story short, for now I'd just handle the cases we have inside Circuit go avoid bloating the scope of this thing. But if you're really attached to finding a solution for other potential use cases we could also do some research (starting with, is this really a problem right now?)

What do you think?

Comment (connor-baer) [reply packages/circuit-ui/components/NotificationInline/NotificationInline.tsx:189]: Sounds good. [KISS](https://www.wikiwand.com/en/KISS_principle).

Comment (robinmetral) [reply packages/circuit-ui/components/NotificationInline/NotificationInline.tsx:189]: Also just stumbled on this when looking into theming, and it looks like Adobe (Spectrum) found an accessible `notify` color. This would probably be the best moving forward (an inaccessible semantic color is always going to cause issues)

Screenshot 2022-01-26 at 17 25 46

File: packages/circuit-ui/components/NotificationToast/NotificationToast.tsx [hunk 2]
```diff
@@ -119,16 +119,37 @@ const contentStyles = ({ theme }: StyleProps) => css`
...
     css`
-      display: block;
+      position: relative;
       align-self: flex-start;
       flex-grow: 0;
       flex-shrink: 0;
       line-height: 0;
       color: ${theme.colors[colorMap[variant]]};
     `,
+  // Adds a black background behind the SVG icon to color just the exclamation mark black.
```
Comment (robinmetral) [inline packages/circuit-ui/components/NotificationToast/NotificationToast.tsx:132]: Did you try to reuse these styles instead of duplicating them across the 3 components? This will make it a bit harder to maintain

Comment (amelako) [reply packages/circuit-ui/components/NotificationToast/NotificationToast.tsx:132]: I looked at it, and I would prefer leaving it as is. The ValidationHint has a slight changes and I think it is not necessary of extracting it as now. If there are similar changes in the future I would rethink the implementation then. I hope that works for you.
I also added the change for the warning text color.

**Review by connor-baer — APPROVED** [review]
💛

File: packages/circuit-ui/components/NotificationToast/NotificationToast.tsx (hunk 2 above)
Comment (robinmetral) [reply packages/circuit-ui/components/NotificationToast/NotificationToast.tsx:132]: Sure that's fine, let's roll with this!

File: packages/circuit-ui/components/ValidationHint/ValidationHint.tsx [hunk 3]
```diff
@@ -43,6 +43,15 @@ const validStyles = ({ theme, showValid }: StyleProps & ValidationHintProps) =>
     color: ${theme.colors.success};
   `;
 
+const warningStyles = ({
+  theme,
+  hasWarning,
+}: StyleProps & ValidationHintProps) =>
+  hasWarning &&
+  css`
+    color: ${theme.colors.n800};
```
Comment (robinmetral) [inline packages/circuit-ui/components/ValidationHint/ValidationHint.tsx:52]: Hm actually I'm not sure this is a good idea, here are my thoughts:

* this means that the `ValidationHint` could be two similar but distinct shades of gray: `n700` (the default, declared in `baseStyles` on line 36) or `n800` (when the warning is there)
* this is adding more complexity to our theme (because right now we can hardcode these colors but semantically, what do they actually mean?)
* if a field with the optional label has a warning, it will have three grays: the label (`n900`), the optional label (`n700`) and the warning (`n800`). This doesn't seem right

So I think we should either embrace our subtle body color (`n700`, the current prod color) or go for `n900` instead to match the label (and in this case, we might want to make this the `ValidationHint` default instead of adding it only to the warning variant).

Comment (robinmetral) [reply packages/circuit-ui/components/ValidationHint/ValidationHint.tsx:52]: Screenshot 2022-02-02 at 16 58 49

Comment (robinmetral) [reply packages/circuit-ui/components/ValidationHint/ValidationHint.tsx]: Fixed in [91af508](https://github.com/sumup-oss/circuit-ui/pull/1386/commits/91af508ffd583b59d4b5728a22c4cf1b73b5c808)

**Review by robinmetral — APPROVED** [review]
Nice! ⚠️
